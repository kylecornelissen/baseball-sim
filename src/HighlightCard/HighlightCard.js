import React from 'react';

import './HighlightCard.scss';

export const HighlightCard = ({highlight, bigVid}) => {
  const featureVid = (
    <video className="highlight-vid" align="left" controls autoplay="autoplay" loop>
      <source src={highlight.videoURL} type="video/mp4"/>
      Your browser does not support the video tag.
    </video>
  )
  const thumbnailImg = (
    <article className="highlight-preview" id={highlight.id}>
      <img
        className="highlight-thumbnail"
        id={highlight.id} src={highlight.thumbnail}
        alt={`${highlight.title}`}
      />
      <h3 className="highlight-title" id={highlight.id}>{highlight.title}</h3>
      <h4 className="highlight-duration" id={highlight.id}>{highlight.time}</h4>
    </article>
  )
  return (
    <article className="highlight-card">
      {!bigVid && thumbnailImg}
      {bigVid && featureVid}
    </article>
  )
}

export default HighlightCard;
