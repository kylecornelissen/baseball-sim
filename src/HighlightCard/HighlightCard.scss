@import '../variables.scss';

.highlight-card {
  display: flex;
  flex-direction: column;
  align-self: center;
  margin: 7.5px 0px;
  width: 95%;
}

.highlight-vid {
  cursor: pointer;
  object-fit: cover;
  outline: none;
  width: 100%;
}

.highlight-preview {
  border: 1px solid $blueaqua;
  cursor: pointer;
  position: relative;
  display: flex;
  width: 100%;
}

.highlight-thumbnail {
  object-fit: fill;
  padding: 0;
  width: 100%;
}

.highlight-title {
  position: absolute;
  top: 5px;
  right: 5px;
  color: $white;
  -webkit-text-fill-color: $white; /* Will override color (regardless of order) */
  -webkit-text-stroke-width: 0.5px;
  -webkit-text-stroke-color: $blueaqua;
  font-size: 1.7vw;
  font-weight: 400;
  padding: 0;
  text-align: right;
  text-shadow: 2px 2px 4px black,
               -2px -2px 4px black,
               -2px 2px 4px black,
               2px -2px 4px black;
  margin: 0px;
  max-width: 94.5%;
  z-index: 100;
}

.highlight-duration {
  color: white;
  background: black;
  margin: 0;
  position: absolute;
  bottom: 5px;
  right: 5px;
}
