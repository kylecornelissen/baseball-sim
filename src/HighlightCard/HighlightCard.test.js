import React from 'react';
import { shallow } from 'enzyme';
import HighlightCard from './HighlightCard';

describe('HighlightCard', () => {
  const highlight = {
    id: 1,
    title: "WOW HOME RUN!",
    videoURL: "https://fakeurl.com/movie1.mp4",
    time: "00:01:30",
    thumbnail: "https://fakeurl.com/movie-pic1.png"
  };

  it('renders a big video if bigVid argument is true', () => {
    const wrapper = shallow(<HighlightCard highlight={highlight} bigVid={true} />);
    const featureVid = <video className="highlight-vid" align="left" controls autoplay="autoplay" loop>
      <source src={highlight.videoURL} type="video/mp4"/>
      Your browser does not support the video tag.
    </video>

    expect(wrapper.contains(featureVid)).toEqual(true);
  });

  it('renders a thumbnail image if bigVid argument is false', () => {
    const wrapper = shallow(<HighlightCard highlight={highlight} bigVid={false} />);

    const thumbnailImg = <article className="highlight-preview" id={highlight.id}>
      <img
        className="highlight-thumbnail"
        id={highlight.id} src={highlight.thumbnail}
        alt={`${highlight.title}`}
      />
      <h3 className="highlight-title" id={highlight.id}>{highlight.title}</h3>
      <h4 className="highlight-duration" id={highlight.id}>{highlight.time}</h4>
    </article>

    expect(wrapper.contains(thumbnailImg)).toEqual(true);
  });

  it('should match the snapshot with all data passed in correctly', () => {
    const wrapper = shallow(<HighlightCard
      highlight={highlight}
      bigVid={false}
    />);

    expect(wrapper).toMatchSnapshot();
  });
});
