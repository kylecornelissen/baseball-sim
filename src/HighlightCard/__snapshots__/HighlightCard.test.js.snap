// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HighlightCard should match the snapshot with all data passed in correctly 1`] = `
<article
  className="highlight-card"
>
  <article
    className="highlight-preview"
    id={1}
  >
    <img
      alt="WOW HOME RUN!"
      className="highlight-thumbnail"
      id={1}
      src="https://fakeurl.com/movie-pic1.png"
    />
    <h3
      className="highlight-title"
      id={1}
    >
      WOW HOME RUN!
    </h3>
    <h4
      className="highlight-duration"
      id={1}
    >
      00:01:30
    </h4>
  </article>
</article>
`;
